include/core/physics_solver.h

```diff
@@ -114,12 +114,36 @@ class PhysicsSolver
    * @brief Default way to evaluate the residual for the nonlinear solver.
```

Comment:
I would use this opportunity to add a better explanation of what is the defualt residual. The default one is the norm of the RHS vector